### ISSUE TITLE
Performance impact throw logging system

### ISSUE THREAD

**Issue by AndreasMager** [moquette-io/moquette#313]
I have started some performance test and identified that the logging system slows down the system a lot.

My setup is 2 clients and 1 server. Each one of them is on an independent machine. 
Client 1 sends 200000 messages on a single topic with QOS 0 and not retained.
Client 2 has subscribed to this topic.

Vernmq(default config) needs 6 seconds to receive and send this messages.

moqutte(default config) needs for this task 18 seconds.

If change both logging level from Info to Error here:
https://github.com/andsel/moquette/blob/master/distribution/src/main/resources/log4j.properties#L5
Than moquette needs only 5 seconds.

Some ideas to address this issue:
* remove logging for each message.
* reduce the log level to debug for message logging.
* update to log4j2 async & gc improved logging

**Comment by windbender** [comment]
@AndreasMager have you done specific profiling to confirm that the performance deficiency is related to logging?  No doubt, efficient logging would be helpful! :-)

**Comment by andsel** [comment]
@AndreasMager looks like Moquette is head to head with VerneMQ ?

**Comment by AndreasMager** [comment]
@windbender I am using Visualvm + jmx to get an overview and information about the hotspots in moquette.  
I found ```Log4jLoggerAdapter.Info``` several times in the combined view with a high total time consumption. 
In the hotspot view it takes also a high rank. So i become curios and tried some different test runs.

@andsel in my opinion moquette wins because VerneMq drops sometimes messages.
My authorization and authentication class are not the default ones. I have used my Cassandra Implementation with a cache.
